lib/std/http/Headers.zig

```diff
@@ -0,0 +1,386 @@
+const std = @import("../std.zig");
+
+const Allocator = std.mem.Allocator;
+
+const testing = std.testing;
+const ascii = std.ascii;
+const assert = std.debug.assert;
+
+pub const HeaderList = std.ArrayListUnmanaged(Field);
+pub const HeaderIndexList = std.ArrayListUnmanaged(usize);
+pub const HeaderIndex = std.HashMapUnmanaged([]const u8, HeaderIndexList, CaseInsensitiveStringContext, std.hash_map.default_max_load_percentage);
+
+pub const CaseInsensitiveStringContext = struct {
+    pub fn hash(self: @This(), s: []const u8) u64 {
+        _ = self;
+        var buf: [64]u8 = undefined;
+        var i: u8 = 0;
+
+        var h = std.hash.Wyhash.init(0);
+        while (i < s.len) : (i += 64) {
+            const left = @min(64, s.len - i);
+            const ret = ascii.lowerString(buf[0..], s[i..][0..left]);
+            h.update(ret);
+        }
+
+        return h.final();
+    }
+
+    pub fn eql(self: @This(), a: []const u8, b: []const u8) bool {
+        _ = self;
+        return ascii.eqlIgnoreCase(a, b);
+    }
+};
+
+pub const Field = struct {
+    name: []const u8,
+    value: []const u8,
+
+    pub fn modify(entry: *Field, allocator: Allocator, new_value: []const u8) !void {
+        if (entry.value.len <= new_value.len) {
+            std.mem.copy(u8, @constCast(entry.value), new_value);
```

Comment:
This `@constCast` does not belong here.

Reply:
I think the reason it gets used here is that the Field might point at non-mutable memory (so value needs to be []const u8), but it also might point to allocator owned memory. This is currently dead code, so for now I'm going to just remove this whole thing, it probably needs to be in Headers anyway.